Problem installing development version with Python bindings

I've (apparently) managed to successfully build the development version of the library with the Python bindings, but when the library is installed I hit an issue. I'm using a non-system Python that is correctly detected, and is not installed in a privileged directory.
```
-- Found PythonInterp: /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/python-2.7.12-r36wznoeqvfh34kbjaqfvd4wmt7pts57/bin/python (found version "2.7.12") 
-- Found PythonLibs: /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/python-2.7.12-r36wznoeqvfh34kbjaqfvd4wmt7pts57/lib/libpython2.7.dylib (found version "2.7.12") 
-- Python interpreter: /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/python-2.7.12-r36wznoeqvfh34kbjaqfvd4wmt7pts57/bin/python
-- Python libraries: /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/python-2.7.12-r36wznoeqvfh34kbjaqfvd4wmt7pts57/lib/libpython2.7.dylib
-- Python include dirs: /Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-7.3.0-apple/python-2.7.12-r36wznoeqvfh34kbjaqfvd4wmt7pts57/include/python2.7
-- Python library version: 2.7.12
```
and the error message is
```
CMake Error at pyranha/cmake_install.cmake:39 (file):
  file cannot create directory: /pyranha.  Maybe need administrative
  privileges.
```
This looks like its trying to install into the root directory?

I'm building on `OSX 10.11 El Capitan`. [Build log](https://github.com/bluescarni/piranha/files/504529/spack-build.txt).

--- a/cmake_modules/yacma/YACMAPythonSetup.cmake
+++ b/cmake_modules/yacma/YACMAPythonSetup.cmake
@@ -25,9 +25,14 @@ set(YACMA_PY_MODULE_EXTENSION "")
 if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     message(STATUS "OS X platform detected.")
     # Apparently on OS X Python expects the .so extension for compiled modules.
-	message(STATUS "Output extension for compiled modules will be '.so'.")
+	  message(STATUS "Output extension for compiled modules will be '.so'.")
     set(YACMA_PY_MODULE_EXTENSION "so")
-    # TODO fill in.
+    # We must establish if the installation dir for Python modules is named 'site-packages' (as usual)
+    # or 'dist-packages' (apparently Ubuntu 9.04 or maybe Python 2.6, it's not clear).
+    execute_process(COMMAND ${PYTHON_EXECUTABLE} "${CMAKE_CURRENT_LIST_DIR}/yacma_python_packages_dir.py"
+        OUTPUT_VARIABLE _YACMA_PY_PACKAGES_DIR OUTPUT_STRIP_TRAILING_WHITESPACE)
+    message(STATUS "Python packages dir is: ${_YACMA_PY_PACKAGES_DIR}")
+    set(YACMA_PYTHON_MODULES_INSTALL_PATH "lib/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/${_YACMA_PY_PACKAGES_DIR}")
 elseif(UNIX)
     message(STATUS "Generic UNIX platform detected.")
     # We must establish if the installation dir for Python modules is named 'site-packages' (as usual)
@@ -41,6 +46,10 @@ elseif(WIN32)
     message(STATUS "Output extension for compiled modules will be '.pyd'.")
     set(YACMA_PY_MODULE_EXTENSION "pyd")
     # TODO fill in.
+endif()
+
+if("${YACMA_PYTHON_MODULES_INSTALL_PATH}" STREQUAL "")
+  message(FATAL_ERROR "Python module install path not detected correctly.")
 endif()
 
 message(STATUS "Python modules install path: " "${YACMA_PYTHON_MODULES_INSTALL_PATH}")

